src/KM_MapEditor.pas

```diff
@@ -249,6 +254,97 @@ procedure TKMMapEditor.UpdateField(aInc: Integer; aCheckPrevCell: Boolean);
 end;
 
 
+procedure TKMMapEditor.EraseObject(aEraseAll: Boolean);
+var Obj: TObject;
+    P: TKMPoint;
+    IsDeleted: Boolean;
+begin
+  IsDeleted := False;

```

Comment:
Let's make it more uniform and with less variables?

For each next object:

if %object_under_cursor% then
begin
  %delete_object%
  if not aEraseAll tehn Exit;
end;

if %next_object_under_cursor% then
 ....

and so on.